django/db/models/fields/__init__.py

```diff
@@ -1539,6 +1540,11 @@ def to_python(self, value):
         if value is None:
             return value
         if isinstance(value, float):
+            if (isnan(value)):
+                raise exceptions.ValidationError(
+                    'NaN value is not supported',
```

Comment:
This string should be in `default_error_messages` and be marked as translatable with the underscore: `_('...')`, although I think it's probably okay to reuse the `'invalid'` message... the message "nan value must be a decimal number." seems to read okay.